theseus/core/loss.py

```diff
@@ -0,0 +1,40 @@
+# Copyright (c) Meta Platforms, Inc. and affiliates.
+#
+# This source code is licensed under the MIT license found in the
+# LICENSE file in the root directory of this source tree.
+
+import abc
+
+import torch
+
+
+class Loss:
```

Comment:
Loss might be a confusing name here given the outer-loop training and learning related terminology.

Also if this is only relevant to `RobustCostFunction` we can probably move all this to that file.

Reply:
How about `LossFunction`?

Reply:
`RobustCost`?
That way `RobustCost` + `CostFunction` -> `RobustCostFunction` would make sense too.